UnityProject/Assets/Scripts/UI/Core/Animations/SillyWalk.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using Core.Physics;
+using HealthV2;
+using Logs;
+using Player.Movement;
+using UnityEngine;
+
+public class SillyWalk : MonoBehaviour
+{
+
+	public float UpAndDownMagnitude = 1;
+	public float RotateMagnitude = 70.88f;
+
+	public LivingHealthMasterBase health;
+	public MovementSynchronisation UOP;
+
+	public void OnEnable()
+	{
+		if (CustomNetworkManager.IsHeadless) return;
+		UpdateManager.Add(CallbackType.UPDATE, UpdateMe );
+	}
+
+	public void OnDisable()
+	{
+		if (CustomNetworkManager.IsHeadless) return;
+		UpdateManager.Remove(CallbackType.UPDATE, UpdateMe );
+	}
+
+	// Update is called once per frame
+    void UpdateMe()
+    {
+	    bool Animate = true;
+
+	    if (health.IsDead || health.IsCrit || health.IsSoftCrit)
+	    {
+		    Animate = false;
+	    }
+
+	    if (UOP.HasSillyWalk == false || UOP.IsCurrentlyFloating || UOP.IsFlyingSliding)
+	    {
+		    Animate = false;
+	    }
+
+	    if (UOP.AllowInput == false || UOP.CurrentMovementType == MovementType.Crawling)
+	    {
+		    Animate = false;
+	    }
+
+	    if (Animate == false)
+	    {
+		    transform.localPosition = Vector3.zero;
```

Comment:
this hard sets everything, which can interfere with other things that set the rotation.

Reply:
nah is new obj

Reply:
alr